src/App.tsx

```diff
@@ -1,7 +1,38 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string | null;
+};
+
+// eslint-disable-next-line
+export class App extends React.Component<{}, State> {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyUp);
+  }
+
+  handleKeyUp = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <p className="App__message">
+          {
+            this.state.pressedKey
+              ? `The last pressed key is [${this.state.pressedKey}]`
```

Comment:
you use it twice, you can destructure it

Reply:
I cannot fix tests, I tried reinstalling npm with cleaning the cache beforehand and it is still not working.

Reply:
It's not a local problem, they don't run on your machine here

Reply:
There is a problem with your code. Subtle, but check what the failed tests say
![image](https://github.com/mate-academy/react_keyboard/assets/71132144/86f9174c-6d29-44ec-b437-a395bfb9683a)